server/log_operation_manager.go

```diff
@@ -218,30 +124,31 @@ type LogOperationManager struct {
 	logOperation LogOperation
 
 	// electionRunner tracks the goroutines that run per-log mastership elections
-	electionRunner      map[int64]*electionRunner
-	pendingResignations chan resignation
-	runnerWG            sync.WaitGroup
-	tracker             *election.MasterTracker
-	heldMutex           sync.Mutex
-	lastHeld            []int64
+	electionRunner map[int64]*election.Runner
+	resignations   chan func()
+	runnerWG       sync.WaitGroup
+	tracker        *election.MasterTracker
+	heldMutex      sync.Mutex
+	lastHeld       []int64
 	// Cache of logID => name; assumed not to change during runtime
 	logNamesMutex sync.Mutex
 	logNames      map[int64]string
 }
 
-// fixupElectionInfo ensures operation parameters have required minimum values.
+// fixupElectionInfo ensures election parameters have required minimum values.
 func fixupElectionInfo(info LogOperationInfo) LogOperationInfo {
-	if info.PreElectionPause < minPreElectionPause {
-		info.PreElectionPause = minPreElectionPause
+	cfg := &info.ElectionConfig
```

Comment:
If the other two parameters (hold interval and spread) were moved into `ElectionConfig` then this could just be passed an election config.

I haven't fully examined the impact of this change though.